index.js

```diff
@@ -429,10 +427,10 @@ CASAuthentication.prototype._handleTicket = function(req, res, next) {
                         req.session[ this.session_info ] = attributes || {};
                     }
                     if (proxyGrantingTicket && this.proxyCallback_url){
-                      this._getPassword(proxyGrantingTicket).then(function (data) {
-                        req.session [ 'password' ] = data;
+                        var pgt = proxyTokens[proxyGrantingTicket];
+                        delete proxyTokens[proxyGrantingTicket];
+                        req.session [ 'pgt' ] = pgt;
```

Comment:
Shouldn't it be the proxy ticket that is set in the session, so:
```suggestion
                        req.session [ 'pt' ] = pgt;
```
```suggestion
                        req.session [ 'pgt' ] = pgt;
```